src/dataset-builder/DatasetBuilder.ts

```diff
@@ -665,12 +671,12 @@ export const DatasetBuilderContents = ({
       requestValid &&
         h(ActionBar, {
           prompt: h(Fragment, [
-            datasetRequestParticipantCount.status === 'Ready'
-              ? formatCount(datasetRequestParticipantCount.state.result.total)
+            snapshotRequestParticipantCount.status === 'Ready'
+              ? formatCount(snapshotRequestParticipantCount.state.result.total)
               : h(Spinner),
             ' participants in this dataset',
           ]),
-          actionText: 'Request access to this dataset',
+          actionText: 'Request this data snapshot',
           onClick: () => setRequestingAccess(true),
```

Comment:
This `onClick` should create the request, I believe. Since we'll need the ID to display on the modal. I imagine we will also want the setter/state for this component to include the ID to pass into the modal.

Reply:
Yep!